### ISSUE TITLE
Kavita OPDS in KOReader returns error: "Cannot get catalog. Server refuses to serve uncompressed content."

### ISSUE THREAD

**Issue by eniad** [Kareadita/Kavita#1678]
**Describe the bug**

I am attempting to get content from Kavita via [KOReader](http://koreader.rocks/). However, I'm getting a "Cannot get catalog. Server refuses to serve uncompressed content." error. This might be incorrect settings on my part rather than a bug. Can I set Kavita to only serve uncompressed content?

**To Reproduce**

1. Install Kavita via Docker
2. Install KOReader on Android: https://github.com/koreader/koreader/releases
3. Connect OPDS in KOReader to Kavita ODPS api endpoint
4. See error "Cannot get catalog. Server refuses to serve uncompressed content."

**Expected behavior**

KOReader should connect to Kavita's OPDS server.

Kavita's ODPS api endpoint works successfully in the Tachiyomi app. I verified both apps are using the same API key in the URL.

KOReader is successfully connected to the Calibre OPDS catalog.

**Devices**

- Kavita host OS: Docker
- Client Device: Android 9
- Clients: KOReader (2022.11), Tachiyomi (0.14.2)

**Additional context**

This error was previously reported by [andycarlberg](https://github.com/andycarlberg) in Kavita's repo as [Kareadita/Kavita#1199](https://github.com/Kareadita/Kavita/issues/1199). In that discussion, an issue was also opened in KOReader's repo as [koreader/koreader#8971](https://github.com/koreader/koreader/issues/8971). In that case the error cleared up on its own. I waited a few days and restarted for client and server multiple times. I also turned on debugging in KOReader to see if that was some how the fix in the original bug report. There was no change in behavior. I also can't find the KOReader log file on Android.

In Kavita, the logs are below. Debug logging level provides the same lines as info.

```log
[Kavita] [2022-12-06 15:43:09.834 -05:00  30] [Information] Serilog.AspNetCore.RequestLoggingMiddleware HTTP HEAD /api/opds/<apikey> responded 200 in 1.6113 ms

[Kavita] [2022-12-06 15:43:09.941 -05:00  30] [Information] Serilog.AspNetCore.RequestLoggingMiddleware HTTP GET /api/opds/<apikey> responded 406 in 86.0732 ms
```

Server and device are on the same local network. Server is not exposed to the internet.

The "Cannot get catalog. Server refuses to serve uncompressed content." is a 406. See [KOReader's opdsbrowser.lua line 302](https://github.com/koreader/koreader/blob/905d5610bacead94f43dfe6bc9bbe70ad471cda0/plugins/opds.koplugin/opdsbrowser.lua#L302).

Internet searches hinted that a 406 error sometimes occurs on pdf > html conversion (although the context was dubious). Removing all pdfs from my Kavita library resulted in no change in behavior.

**Comment by majora2007** [comment]
If I recall, the previous user's error was due to their reverse proxy compressing response to KOReader.  From the logs attached, the server is responding with valid output, this seems more like something KOReader should be looking into. Kavita is serving valid OPDS Spec XML, hence why the other readers work fine.

**Comment by majora2007** [comment]
I just downloaded KOReader and added my OPDS url and it's working 100%. Are you sure you're following our instructions and using JUST your OPDS url from your user settings?

**Comment by eniad** [comment]
It's working now! Thank you for the support. I figured it was something I over looked.

---

I did another read of the instructions at https://wiki.kavitareader.com/en/guides/settings/opds

I had skipped the turning on of the OPDS for two reasons:

1. It worked in another App
2. The instructions pointed me to the wrong page

I made an assumption that the documentation was out of date. Sorry for that assumption!
